networkx/classes/digraph.py

```diff
@@ -43,6 +46,9 @@ def __set__(self, obj, value):
             del od["adj"]
         if "succ" in od:
             del od["succ"]
+        for prop in ["edges", "out_edges", "degree", "out_degree", "in_degree"]:
+            if prop in od:
+                del od[prop]
```

Comment:
It looks like `"adj"` and `"succ"` could just be added to this list to save a few lines, but there's no reason to do so other than style. Just thought I'd mention!